pkg/debug/networkchaos/networkchaos.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2019 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package networkchaos
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/chaos-mesh/chaos-mesh/pkg/debug/common"
+)
+
+func Debug(chaos string, ns string) error {
+	chaosList, err := common.Debug("networkchaos", chaos, ns)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+	for _, chaosName := range chaosList {
+		fmt.Println(string(common.ColorCyan), "[CHAOSNAME]:", chaosName, string(common.ColorReset))
+		if err := debugEachChaos(chaosName, ns); err != nil {
+			return fmt.Errorf("debug chaos failed with: %s", err.Error())
+		}
+	}
+	return nil
+}
+
+func debugEachChaos(chaosName string, ns string) error {
+	p, err := common.GetPod("networkchaos", chaosName, ns)
+	if err != nil {
+		return err
+	}
+
+	// get nsenter path from log
+	var nsenterPathList []string
+	for _, tailNum := range []int64{100, 1000, 10000, -1} {
+		log, err := common.GetLog(p.ChaosDaemonName, p.ChaosDaemonNamespace, tailNum)
+		if err != nil {
+			return fmt.Errorf("get log failed with: %s", err.Error())
+		}
+		nsenterPathList = regexp.MustCompile("(?:-n/proc/)(.*)(?:/ns/net)").FindStringSubmatch(log)
```

Comment:
I don't think you should get this information from log, which is obviously unstable.

You can get this information from `chaos-daemon`. It can lookup PID through the container id, and the namespace path is `/proc/$PID/ns/...`

However, communicating with `chaos-daemon` could be not that easy for a command line tool (outside the cluster). A possible solution is to place this binary in controller-manager image, and execute it. 

Or setup a `kubectl port-forward` (or directly import the go pkg and call it from your code, if it's simple enough.) in background to communicate with the `chaos-daemon`.

Reply:
Thanks for the suggestions! I'll look into it

Reply:
fixed. 
Just found out there is already an implementation of port-forward for [e2e test](https://github.com/chaos-mesh/chaos-mesh/blob/master/test/e2e/util/portforward/portforward.go), so the work is easier than expected.

Should I move this `portforward` folder to pkg/utils, since it's not only used by e2e anymore?